src/hotspot/share/runtime/mutex.cpp

```diff
@@ -207,23 +216,22 @@ bool Monitor::wait_without_safepoint_check(int64_t timeout) {
 
 bool Monitor::wait(int64_t timeout, bool as_suspend_equivalent) {
   JavaThread* const self = JavaThread::current();
+  // Safepoint checking logically implies an active JavaThread.
+  guarantee(self->is_active_Java_thread(), "invariant");
```

Comment:
Not clear why this is necessary here, and why it is a guarantee rather than assert ??

Reply:
I looked at the history of this check. The guarantee has been since day1 on the hg repo as "self->is_Java_thread()" and then changed in 8226228 to "self->is_active_Java_thread()". I moved it at the beginning of wait() since it seems the right place to check it just after retrieving self. It doesn't make sense to keep it as a guarantee when every other check is an assert so I will change it.